addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -1419,10 +1494,28 @@ MockServer.include({
             const trackingValueIds = this._getRecords('mail.tracking.value', [
                 ['id', 'in', message.tracking_value_ids],
             ]);
+            const message_reactions = this._getRecords('mail.message.reaction', [
+                ['message_id', '=', message.id],
+            ]);
+            let messageReactionGroups = [];
+            for(const reaction of message_reactions) {
+                let index = messageReactionGroups.findIndex(a_reaction => a_reaction.content === reaction.content);
```

Comment:
If you need to work with index like that, you are probably better off working with a `Set` instead (and convert to array before returning the result).

Reply:
`Set` or `Map` probably as you likely need key/value